<launch>
    <arg name="init_x" value="0.0"/>
    <arg name="init_y" value="0.0"/>
    <arg name="init_z" value="1.0"/> 
    
    <arg name="index" default="0" />

    <node pkg="so3_quadrotor_simulator" type="quadrotor_simulator_so3" name="quadrotor_simulator_controller_$(arg index)" output="screen">
        <param name="rate/odom" value="50.0"/>
        <param name="rate/simulation" value="30.0"/>
        <param name="simulator/init_state_x" value="$(arg init_x)"/>
        <param name="simulator/init_state_y" value="$(arg init_y)"/>
        <param name="simulator/init_state_z" value="$(arg init_z)"/>
        <remap from="~odom" to="quad_$(arg index)/sim/odom"/>
        <remap from="~cmd" to="quad_$(arg index)/cmd"/>
        <remap from="~reset" to="quad_$(arg index)/reset"/>
        <remap from="~uav" to="quad_$(arg index)/uav"/>
        <remap from="~imu" to="/sim/imu"/>
        <remap from="~force_disturbance" to="force_disturbance"/>    
        <remap from="~moment_disturbance" to="moment_disturbance"/>        
    </node>

</launch>
